modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -199,7 +199,9 @@ public ObjectDefinition addObjectDefinition(
 			ObjectEntryTable.INSTANCE.objectEntryId.getName(),
 			objectDefinition.isSystem(), userId);
 
-		return objectDefinition;
+		_setTitleObjectFieldId(objectDefinition, StringPool.BLANK);
+
+		return objectDefinitionPersistence.update(objectDefinition);
```

Comment:
Because, we need to have the **ObjectDefinition** persisted to create its **ObjectFields**, and is needed all of these ObjectFields persisted, to get some of them and set with **titleObjectFieldId**. 

So, I have no ideia to work around this two `updates`

Reply:
@p-albuquerque got it.. no problem.. I recommend adding a comment in the code so that Brian understands why we did it this way.